frontend/src/app/search.tsx

```diff
@@ -395,8 +394,14 @@ class Search extends Component<IAppState, ISearchState> {
 
   private sort(key: string) {
     // Switch the direction each time some clicks.
-    this.setState({ isSortedUp: !this.state.isSortedUp, sortKey: key }, () => {
+    this.setSortedUp()
+    this.setState({sortKey: key }, () => {
       this.getSortedRows()
+    }) 
```

Comment:
This is a small issue - you're calling `setSortedUp()` which is a `setState()` followed by `getSortedRows()`.
Then, you're performing your own `setState()` followed by another `getSortedRows()`.
This results in two rerenders and two fetches (verified in browser network console).
Can you combine both duplicates?